arangod/RocksDBEngine/RocksDBV8Functions.cpp

```diff
@@ -298,7 +298,7 @@ static void JS_CollectionRevisionTreeVerification(
     try {
       auto* physical = toRocksDBCollection(*collection);
       auto batchId = ctx->id();
-      storedTree = physical->revisionTree(batchId);
+      storedTree = physical->revisionTree(ctx->snapshot()->GetSequenceNumber());
```

Comment:
Again, in theory the `snapshot()` function can return a nullptr, and we should check the return value here before dereferencing it.

Reply:
Good point, I have added a check.